src/server/server_family.cc

```diff
@@ -978,6 +975,53 @@ void ServerFamily::Shutdown() {
   });
 }
 
+void ServerFamily::UpdateMemoryGlobalStats() {
+  ShardId sid = EngineShard::tlocal()->shard_id();
+  if (sid != 0) {  // This function is executed periodicaly on all shards. To ensure the logic
+                   // bellow runs only on one shard we return is the shard is not 0.
+    return;
+  }
+  time_t curr_time = time(nullptr);
+  if (curr_time == global_stats_update_time_) {  // Runs one a second.
+    return;
+  }
+  global_stats_update_time_ = curr_time;
+
+  uint64_t mem_current = used_mem_current.load(std::memory_order_relaxed);
+  if (mem_current > used_mem_peak.load(memory_order_relaxed)) {
+    used_mem_peak.store(mem_current, memory_order_relaxed);
+  }
+
+  io::Result<io::StatusData> sdata_res = io::ReadStatusInfo();
+  if (sdata_res) {
+    size_t total_rss = sdata_res->vm_rss + sdata_res->hugetlb_pages;
+    rss_mem_current.store(total_rss, memory_order_relaxed);
+    if (rss_mem_peak.load(memory_order_relaxed) < total_rss) {
+      rss_mem_peak.store(total_rss, memory_order_relaxed);
+    }
+    double rss_oom_deny_ratio = absl::GetFlag(FLAGS_rss_oom_deny_ratio);
```

Comment:
you added `rss_oom_deny_ratio` to server state but here you use the flag.

Reply:
will fix